### PR TITLE
Added icon readme and js doc for icon

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # .

#### Changes proposed in this Pull Request:
- Added icon readme
- Added js doc for icon
- 

#### Screenshots / Screen Recording (if applicable)
-
Screenshot 2023-12-22 at 3 55 32 PM

#### To Test:
- 

#### Any specific feedback you are looking for?
-

### CODE REVIEW

**Review by huayunh — COMMENTED** [review]
Now we have an `Icon.md` and `Icons.md`. I would not know when to use what. Consider merging them together -- "There are two ways you can include icons in your application. The first is... Alternatively, if you... consider using..."

**Review by ektaghag-eaton — APPROVED** [review]

File: docs/Icons.md [hunk 1]
```diff
@@ -72,28 +76,15 @@ import { Battery } from '@brightlayer-ui/react-native-progress-icons';
 
 Brightlayer UI components will pass these values to you for use in your component to achieve the correct appearance. It is your responsibility to apply them.
 
-## Wrapped Icon
```
Comment (ektaghag-eaton) [inline docs/Icons.md:75]: You have made same changes here https://github.com/etn-ccis/blui-react-native-component-library/pull/483/files#diff-b1b786098c4fa4ff2f6eea4aa4816179b09676a5b43c6b2f3c3f6920dc9bfe7bL76.

File: docs/Icons.md [hunk 2]
```diff
@@ -102,11 +93,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
...
 The icon prop will also support images in various formats, either from local resources or from the web.
 
 ```tsx
 <Component icon={{ uri: 'https://raw.githubusercontent.com/etn-ccis/blui-icons/dev/packages/png/png48/account_settings_black500_48dp.png' }} />
 <Component icon={require('./path/to/image.png')} />
 ```
+
+## Icon as a component
+
+`<Icon>` component is an internal utility component used to render icons inside of other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside of every component that utilizes icons.
```
Comment (huayunh) [inline docs/Icons.md:107]:
```suggestion
`<Icon>` component is an internal utility component used to render icons inside other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside every component that utilizes icons.
```

File: docs/Icons.md [hunk 3]
```diff
@@ -102,11 +93,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
...
+
+<img width="100%" style="max-width: 300px; display: block;" alt="Icon" src="./images/icon.png">
```
Comment (huayunh) [inline docs/Icons.md:109]: this image is huge

File: docs/Icons.md [hunk 4]
```diff
@@ -102,11 +93,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
...
+<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
+```
+
+### API
+
+<div style="overflow: auto;">
+
+| Prop Name | Description                        | Type                                   | Required | Default                   |
+| --------- | ---------------------------------- | -------------------------------------- | -------- | ------------------------- |
+| source    | A component to render for the icon | `IconSource`                           | yes      |                           |
```
Comment (huayunh) [inline docs/Icons.md:125]: Is "IconSource" a type the same as "Icon Object"'s API? Or are you expecting your user to pass a component, as suggested by the readme?

Comment (surajeaton) [reply docs/Icons.md:125]: Yes IconSource type is same as icon object

Comment (huayunh) [reply docs/Icons.md]: You are still saying "A component to render for the icon" here, which suggests to me that you are expecting adopters to pass something that looks like `<IconComponent ... />` for the `source` prop

File: docs/Icons.md [hunk 5]
```diff
@@ -103,11 +94,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
...
+<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
+```
+
+### API
+
+<div style="overflow: auto;">
+
+| Prop Name | Description                                                     | Type                                   | Required | Default                   |
+| --------- | --------------------------------------------------------------- | -------------------------------------- | -------- | ------------------------- |
+| source    | An object has the image source or an icon family or JSX.Element | [`IconSource`](#icon-object)           | yes      |                           |
```
Comment (huayunh) [inline docs/Icons.md:126]: you will want to 
* not write this a sentence, but a description of `source`
* make it succinct 

```suggestion
| source    | An object to specify the source of the icon | [`IconSource`](#icon-object)           | yes      |                           |
```

File: components/src/core/Icon/Icon.tsx [hunk 6]
```diff
@@ -10,7 +10,9 @@ import { calculateHeight } from '../Utility/shared';
 import { ExtendedTheme, useExtendedTheme } from '@brightlayer-ui/react-native-themes';
 
 export type IconProps = IconComponentProps & {
+    /** A component to render for the icon */
```
Comment (huayunh) [inline components/src/core/Icon/Icon.tsx:13]: update this too

**Review by huayunh — APPROVED** [review]

**Pull request merged**